crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1060,31 +1067,26 @@ impl CumulocityConverter {
     }
 
     fn try_init_messages(&mut self) -> Result<Vec<Message>, ConversionError> {
-        let inventory_fragments_message = self.wrap_error(create_inventory_fragments_message(
-            &self.device_name,
-            &self.cfg_dir,
-        ));
+        let mut messages = vec![];
+
+        let mut inventory_fragments_message = self.create_base_inventory_message()?;
+        messages.append(&mut inventory_fragments_message);
 
         let supported_operations_message = self.wrap_error(
             self.create_supported_operations(&self.cfg_dir.join("operations").join("c8y")),
         );
+        messages.push(supported_operations_message);
 
-        let cloud_child_devices_message = create_request_for_cloud_child_devices();
+        let device_data_message = self.inventory_device_type_update_message()?;
+        messages.push(device_data_message);
 
-        let device_data_message = self.wrap_error(create_device_data_fragments(
-            &self.device_name,
-            &self.device_type,
-        ));
+        let pending_operations_message = create_get_pending_operations_message()?;
+        messages.push(pending_operations_message);
 
-        let pending_operations_message = self.wrap_error(create_get_pending_operations_message());
+        let cloud_child_devices_message = create_request_for_cloud_child_devices();
+        messages.push(cloud_child_devices_message);
 
-        Ok(vec![
-            inventory_fragments_message,
-            supported_operations_message,
-            device_data_message,
-            pending_operations_message,
-            cloud_child_devices_message,
-        ])
+        Ok(messages)
```

Comment:
Yeah, I was forced into this change because  `inventory_fragments_message` is not a `Message` anymore but a `Vec<Message>`. To restore the old readability, I'll keep the rest of the messages in this `vec![]` itself and then merge it with `inventory_fragments_messages` right before returning.